ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/request/ApplicationReceiveFunctions.kt

```diff
@@ -58,7 +58,8 @@ public suspend inline fun <reified T : Any> ApplicationCall.receiveOrNull(): T?
  * @return instance of [T] received from this call.
  * @throws ContentTransformationException when content cannot be transformed to the requested type.
  */
-public suspend inline fun <reified T : Any> ApplicationCall.receive(): T = receiveNullable(typeInfo<T>())!!
+public suspend inline fun <reified T : Any> ApplicationCall.receive(): T = receiveNullable(typeInfo<T>())
+    ?: throw CannotTransformContentToTypeException(typeInfo<T>().kotlinType!!)
```

Comment:
Also `CannotTransformContentToTypeException` will be thrown if transformation result is `null` and it's successful.

Reply:
Why do need to throw the same exception when it's raised already in `ApplicationCall.receiveNullable`? The only case it covers is when a result is `NullBody` so the exception is misleading.

Reply:
We throw `ContentTransformationException` if content cannot be transformed to the requested type. If content is converted to null for non nullable type it's `ContentTransformationException`